pexpect/utils.py

```diff
@@ -154,3 +154,32 @@ def select_ignore_interrupts(iwtd, owtd, ewtd, timeout=None):
                 # something else caused the select.error, so
                 # this actually is an exception.
                 raise
+
+
+def poll_ignore_interrupts(fds, timeout=None):
+    """Simple wrapper around poll for a list of file descriptors."""
+
+    if timeout is not None:
+        end_time = time.time() + timeout
+
+    poller = select.poll()
+    for fd in fds:
+        poller.register(fd)
+        while True:
```

Comment:
This should probably be outside the for loop, if I'm understanding it correctly.

Reply:
Before I merge this, can you confirm that this is what you want to do?

Reply:
I want to register each fd right or am I misunderstanding what I need to do here? I guess we're only really ever passing 1 FD tho ...

Reply:
you never break the while loop and it has no condition to allow the next `fd` to come into play.

Reply:
I think the `while` loop should be dedented one level so it happens after the `for` loop instead of inside it.

It doesn't make a difference when we only pass one fd, but it will be broken if the function is ever used with more than one, so we might as well get it right.